go/tasks/pluginmachinery/remote/plugin.go

```diff
@@ -0,0 +1,81 @@
+package remote
+
+import (
+	"context"
+
+	"github.com/lyft/flytestdlib/promutils"
+
+	pluginsCore "github.com/lyft/flyteplugins/go/tasks/pluginmachinery/core"
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/io"
+)
+
+//go:generate mockery -all -case=underscore
+
+// A Lazy loading function, that will load the plugin. Plugins should be initialized in this method. It is guaranteed
+// that the plugin loader will be called before any Handle/Abort/Finalize functions are invoked
+type PluginLoader func(ctx context.Context, iCtx PluginSetupContext) (Plugin, error)
+
+// PluginEntry is a structure that is used to indicate to the system a K8s plugin
+type PluginEntry struct {
+	// ID/Name of the plugin. This will be used to identify this plugin and has to be unique in the entire system
+	// All functions like enabling and disabling a plugin use this ID
+	ID pluginsCore.TaskType
+
+	// A list of all the task types for which this plugin is applicable.
+	SupportedTaskTypes []pluginsCore.TaskType
+
+	// An instance of the plugin
+	PluginLoader PluginLoader
+
+	// Boolean that indicates if this plugin can be used as the default for unknown task types. There can only be
+	// one default in the system
+	IsDefault bool
+}
+
+type PluginSetupContext interface {
+	// a metrics scope to publish stats under
+	MetricsScope() promutils.Scope
+	// Returns a secret manager that can retrieve configured secrets for this plugin
+	SecretManager() pluginsCore.SecretManager
+}
+
+type TaskExecutionContext interface {
+	// Returns a TaskReader, to retrieve task details
+	TaskReader() pluginsCore.TaskReader
+
+	// Returns an input reader to retrieve input data
+	InputReader() io.InputReader
+
+	// Returns a handle to the Task's execution metadata.
+	TaskExecutionMetadata() pluginsCore.TaskExecutionMetadata
+
+	// Provides an output sync of type io.OutputWriter
+	OutputWriter() io.OutputWriter
+}
+
+// The resource to be sycned from the remote
+type ResourceMeta = interface{}
```

Comment:
Why is this called "Meta"?  What is meta about it?  It seems like it's the resource itself, not metadata about the resource.

Reply:
You're right, it's actually metadata about the resource yes.